### DISCUSSION TITLE
Camera Config file for the Dahua IPC-T5442TM-AS

### DISCUSSION THREAD

**Discussion by dz4news** [blakeblackshear/frigate#4114] [Ask A Question]
Can anyone that is using one of these cameras share the config they are using for it? I can get the camera working using VLC, but everytime I try to add it to Frigate it blows up. I have a couple of Reolink cameras working, but I dont like the night vision they have so  I am trying to use one of the dahua cameras.
Thanks..

**Comment by NickM-27** [comment]
Just paste the logs that you're getting and the config you're using.

**Reply by dz4news** [reply]
I cant seem to get this camera added again... what am I doing wrong? I have 1 reolink camera working great using RTMP.. but for some reason this is the only camera I can add.. every time I add a second camera config, all I get is errors.
Here is the log I currently see..
The camera I am adding is called CARPORT.. 
I dont understand what extra field it is referring too.. since the config format is identical to the camera that is working... with the exception of the path...
***    Config Validation Errors                           ***
*************************************************************
1 validation error for FrigateConfig
record -> CARPORT
  extra fields not permitted (type=value_error.extra)
Traceback (most recent call last):
  File "/opt/frigate/frigate/app.py", line 332, in start
    self.init_config()
  File "/opt/frigate/frigate/app.py", line 82, in init_config
    user_config = FrigateConfig.parse_file(config_file)
  File "/opt/frigate/frigate/config.py", line 942, in parse_file
    return cls.parse_obj(config)
  File "/usr/local/lib/python3.9/dist-packages/pydantic/main.py", line 521, in parse_obj
    return cls(**obj)
  File "/usr/local/lib/python3.9/dist-packages/pydantic/main.py", line 341, in __init__
    raise validation_error
pydantic.error_wrappers.ValidationError: 1 validation error for FrigateConfig
record -> CARPORT
  extra fields not permitted (type=value_error.extra)
*************************************************************
***    End Config Validation Errors                       ***

**Reply by NickM-27** [reply]
Can't do anything without seeing your config.

**Reply by dz4news** [reply]
ok... give me a min.

**Reply by dz4news** [reply]
Here is my config... The camera named "Backporch" works great... once I add the second camera.. everything fails.. I did remove the MQTT info that is at the very top...
      
    detectors:
      coral:
        type: edgetpu
        device: usb
        
    birdseye:
      enabled: True
      width: 1280
      height: 720
      quality: 15
      mode: motion
      
    cameras:
      BACKPORCH:
        ffmpeg:
          input_args: -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -flags low_delay -strict experimental -analyzeduration 1000M -probesize 1000M -rw_timeout 5000000
          output_args:
            record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy
          inputs:
            - path: rtmp://<camera ip>/bcs/channel0_main.bcs?channel=0&stream=1&user=username&password=password
              roles:
                - rtmp
            - path: rtmp://<camera ip>/bcs/channel0_sub.bcs?channel=0&stream=1&user=username&password=password
              roles:
                - detect
                - record
    detect:
      width: 640
      height: 480
      fps: 7
      
    objects:
      track:
        - person
        - dog
        - cat
        - bird
    snapshots:
      enabled: true
      timestamp: false
      bounding_box: true
      crop: false
      retain:
        default: 5
    record:
      enabled: true
      events:
        pre_capture: 3
        post_capture: 3
        objects:
          - person
          - dog
          - cat
        retain:
          default: 3
      
    motion:
      mask:
        - 206,36,523,30,532,56,572,66,640,62,640,0,204,0

    CARPORT:
      ffmpeg:
        input_args: -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -flags low_delay -strict experimental -analyzeduration 1000M -probesize 1000M -rw_timeout 5000000
        output_args:
          record: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy
        inputs:
          - path: rtmp://<camera ip>/bcs/channel1_main.bcs?channel=0&stream=1&user=username&password=password
            roles:
              - rtmp
          - path: rtmp://<camera ip>/bcs/channel1_sub.bcs?channel=0&stream=1&user=username&password=password
            roles:
              - detect
              - record
      detect:
        width: 640
        height: 480
        fps: 7
        
        objects:
        track:
          - person
          - dog
          - cat
          - bird
      snapshots:
        enabled: true
        timestamp: false
        bounding_box: true
        crop: false
        retain:
        default: 5
      record:
        enabled: true
        events:
          pre_capture: 3
          post_capture: 3
      objects:
        - person
        - dog
        - cat
      retain:
        default: 3

**Reply by dz4news** [reply]
This is the most current log file... using the config above...
***    Config Validation Errors                           ***
*************************************************************
1 validation error for FrigateConfig
CARPORT
  extra fields not permitted (type=value_error.extra)
Traceback (most recent call last):
  File "/opt/frigate/frigate/app.py", line 332, in start
    self.init_config()
  File "/opt/frigate/frigate/app.py", line 82, in init_config
    user_config = FrigateConfig.parse_file(config_file)
  File "/opt/frigate/frigate/config.py", line 942, in parse_file
    return cls.parse_obj(config)
  File "/usr/local/lib/python3.9/dist-packages/pydantic/main.py", line 521, in parse_obj
    return cls(**obj)
  File "/usr/local/lib/python3.9/dist-packages/pydantic/main.py", line 341, in __init__
    raise validation_error
pydantic.error_wrappers.ValidationError: 1 validation error for FrigateConfig
CARPORT
  extra fields not permitted (type=value_error.extra)
*************************************************************
***    End Config Validation Errors                       ***
*************************************************************

**Reply by NickM-27** [reply]
CARPORT and everything under it needs to be moved up otherwise it's not a part of the `cameras` field

**Reply by dz4news** [reply]
alright... I will give that a try. Thanks

**Reply by dz4news** [reply]
Looks like that worked... I have two cameras working now.

I will get the dahua working again and reach out if I need help. Thanks for the assistance.

**Comment by dz4news** [comment]
I finally got my Dahua IPC-T5442TM-AS working in Frigate... Below is my current config for it. I am really struggling trying to find the correct settings for it in Frigate.
While It is working, I know it can be much better.
My other cameras are Reolinks and this thing has incredible low light vision... so much that I thought it was hung on a daylight image, until I walked in front of it after dark.
Any help with tweaking the config is appreciated.

```backyard:
    ffmpeg:
      inputs:
        - path: rtsp://user:pass@192.168.x.xxx:554/cam/realmonitor?channel=1&subtype=2
          roles:
            - detect
        - path: rtsp://user:pass@192.168.x.xxx:554/cam/realmonitor?channel=1&subtype=0
          roles:
            - record
            - rtmp
    detect:
      width: 1280
      height: 720
    objects:
      track:
        - person
        - dog
        - cat
    motion:
      mask:
        - 0,166,361,104,513,103,637,87,1116,100,1280,0,0,0```

**Reply by NickM-27** [reply]
What do you think can be better? Most cameras don't require the tweaking or anything else like reolinks do

**Reply by dz4news** [reply]
Coming from ReoLink with a crazy amount of tweaks and finding very little info on these cameras in Frigate, I expected some amount of tweaks, etc. Really did not think it was that easy...
While it does work, the reolink e1 outdoor seemed to have better day time detection, just terrible night vision. That is where I was thinking some extra settings would help.
What about the main and sub streams? it has a main and two sub streams... am I calling them correct?
Thanks for your assistance to all my questions... I usually learn from examples and then adjust them to my scenarios..

**Reply by NickM-27** [reply]
You'll need to do some looking to understand why it seems daytime detection isn't working as well. Try watching the debug view in frigate with bounding boxes enabled

**Comment by edkoch25** [comment]
DZ did you ever get things working? 

I'm running into similar issues and would love to know what your final config looked like.

**Comment by Danielson89** [comment]
I just discovered that visiting the sun and standing on it is far more achievable than configuring this ffs.  Sorry I've dropped this silly program and installed Milestone,  took me around 10 minutes to configure it for 8 Dahua cameras.  Yeah I just don't understand why it's actually impossible yes impossible to use this,  taken about 3 months of mucking around and it still doesn't work.

HomeAssistant has a plugin for Dahua, 
Milestone X-Protect is free up to  8 cameras,  I recommend it,  no more pain

**Comment by dz4news** [comment]
@edkoch25 Sorry I did not see your post sooner... but here is a portion of my config for the 6 Dahua cameras I have running in Frigate.
The portion of the config is just for 1 camera... but you can add to the go2rtc for additional cameras and then add to the cameras: config..

The system is running very smooth and stable 

```....
#### go2rtc config ###
go2rtc:
  streams:
    frontnorth:
      - rtsp://user:pass@192.168.0.69:554/live0
      - "ffmpeg:frontnorth#audio=opus"
    frontnorth_sub: rtsp://user:pass@192.168.0.69:554/cam/realmonitor?channel=1&subtype=2
ffmpeg:
  global_args: -hide_banner -loglevel warning -threads 2
  hwaccel_args: preset-vaapi
  input_args: preset-rtsp-restream
  output_args:
    detect: -threads 2 -f rawvideo -pix_fmt yuv420p
    record: preset-record-generic-audio-copy
motion:
  threshold: 25
  improve_contrast: false
### Camera Settings ###
### Front Camera - Facing North IPC-T5442TM-AS 2.8mm ###
cameras:
  frontnorth:
    enabled: true
    audio:
      enabled: true
    ui:
      order: 0
    birdseye:
      order: 0
    ffmpeg:
      inputs:
      - path: rtsp://192.168.0.227:8554/frontnorth_sub
        roles:
          - audio
          - detect
      - path: rtsp://192.168.0.227:8554/frontnorth
        roles:
          - record
    objects:
      filters:
        bird:
          min_score: 0.85
        car:
          min_area: 2000
          max_area: 24000000
          min_score: 0.85
        amazon:
          min_score: 0.75
        fedex:
          min_score: 0.75
        ups:
          min_score: 0.75
        cat:
          min_score: 0.90
          threshold: 0.7
        dog:
          min_score: 0.90
          threshold: 0.7
        deer:
          min_score: 0.85
          threshold: 0.7
        person:
          min_score: 0.90
          threshold: 0.7
    motion:
      mask:
        - 528,720,525,665,0,673,0,720
        - 1280,0,298,0,148,144
    zones:
      zone_frontyard:
        coordinates: 476,174,1280,371,1280,93,879,49,379,127,140,188,143,238
        objects:
          - person
          - dog
          - deer
      zone_front_patio:
        coordinates: 0,720,1280,720,1280,398,846,264,467,190,39,273
        objects:
          - person
          - dog
          - deer
          - cat
```
